routers/web/repo/issue.go

```diff
@@ -2038,7 +2099,7 @@ func NewComment(ctx *context.Context) {
 		return
 	}
 
-	if !ctx.IsSigned || (ctx.User.ID != issue.PosterID && !ctx.Repo.CanReadIssuesOrPulls(issue.IsPull)) {
+	if !ctx.IsSigned || (ctx.User.ID != issue.PosterID && !ctx.Repo.CanReadIssuesOrPulls(issue.IsPull)) || issue.IsPrivate && !(ctx.User.ID != issue.PosterID || ctx.Repo.CanReadPrivateIssues()) {
```

Comment:
```suggestion
	if !ctx.IsSigned ||
		(ctx.User.ID != issue.PosterID && !ctx.Repo.CanReadIssuesOrPulls(issue.IsPull)) ||
		(issue.IsPrivate && !(ctx.User.ID != issue.PosterID || ctx.Repo.CanReadPrivateIssues())) {
```

Let's avoid some potential confusion here and make this easier to read.